src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private int size;
+    private int capacity;
+    private float loadFactor;
+    private Node<K, V>[] buckets;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY, LOAD_FACTOR);
+    }
+
+    public MyHashMap(int capacity, float loadFactor) {
+        this.capacity = capacity;
+        this.loadFactor = loadFactor;
+        this.buckets = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int bucketIndex = getBucketIndex(key);
+        Node<K, V> node = buckets[bucketIndex];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value, buckets[bucketIndex]);
+        buckets[bucketIndex] = newNode;
+        size++;
+        if (needsResize()) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int bucketIndex = getBucketIndex(key);
+        Node<K, V> node = buckets[bucketIndex];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getBucketIndex(K key) {
+        int hash = Objects.hashCode(key);
+        hash = hash < 0 ? -hash : hash;
+        int index = hash % capacity;
+        return index < 0 ? index + capacity : index;
+    }
+
+    private boolean needsResize() {
+        float currentLoadFactor = (float) size / capacity;
+        return currentLoadFactor > loadFactor;
+    }
+
+    private void resize() {
+        if (size + 1 > buckets.length * LOAD_FACTOR) {
```

Comment:
You have initialized capacity but use bucket.length ?

Reply:
it is safe to use
 so i just cant go over tests